src/ol/layer/STAC.js

```diff
@@ -0,0 +1,625 @@
+
+    /**
+     * @type {GeoTIFFSourceOptions}
+     */
+    let options = {
+      sources: [sourceInfo],
+    };
+
+    const projection = await getProjection(asset);
+    if (projection) {
+      options.projection = projection;
+    }
+
+    if (this.getGeoTIFFSourceOptions_) {
+      options = await this.getGeoTIFFSourceOptions_(options, asset);
+    }
+
+    const tileserverFallback = async (asset, layer) => {
+      if (layer) {
+        this.getLayers().remove(layer);
+      }
+      return await this.addTileLayer_(asset);
+    };
+    try {
+      const source = new GeoTIFF(options);
+      const layer = new WebGLTileLayer({source});
+      if (this.useTileLayerAsFallback_) {
+        const errorFn = () => tileserverFallback(asset, layer);
+        source.on('error', errorFn);
+        source.on('tileloaderror', errorFn);
+        // see https://github.com/openlayers/openlayers/issues/14926
+        source.on('change', () => {
+          if (source.getState() === 'error') {
+            tileserverFallback(asset, layer);
+          }
+        });
+        layer.on('error', errorFn);
+        // Call this to ensure we can load the GeoTIFF, otherwise try fallback
+        await source.getView();
+      }
+      this.addLayer_(layer, asset);
+      return layer;
+    } catch (error) {
+      if (this.useTileLayerAsFallback_) {
+        return await tileserverFallback(asset, null);
+      }
+      this.handleError_(error);
+    }
+  }
+
+  /**
+   * @private
+   * @param {Asset|Link} [data] A STAC Asset or Link
+   * @return {Promise<TileLayer>} Resolves with a TileLayer when complete.
+   */
+  async addTileLayer_(data) {
+    /**
+     * @type {XYZSourceOptions}
+     */
+    let options = {
+      crossOrigin: this.crossOrigin_,
+      url: this.buildTileUrlTemplate_(data),
+    };
+    if (this.getXYZSourceOptions_) {
+      options = await this.getXYZSourceOptions_(options, data);
+    }
+    const layer = new TileLayer({
+      source: new XYZ(options),
+    });
+    this.addLayer_(layer, data);
+    return layer;
+  }
+
+  /**
+   * @param {Layer|LayerGroup} [layer] A Layer to add to the LayerGroup
+   * @param {STAC|Asset} [data] The STAC data source
+   * @param {number} [zIndex=0] The z-index for the layer
+   * @private
+   */
+  addLayer_(layer, data, zIndex = 0) {
+    layer.set('stac', data);
+    layer.setZIndex(zIndex);
+    this.getLayers().push(layer);
+  }
+
+  /**
+   * @private
+   * @return {VectorLayer|null} The vector layer showing the geometry/bbox.
+   */
+  addFootprint_() {
+    let geojson = null;
+    const data = this.getData();
+    if (data.isItemCollection() || data.isCollectionCollection()) {
+      geojson = toGeoJSON(data.getBoundingBox());
+    } else {
+      geojson = data.toGeoJSON();
+    }
+
+    if (geojson) {
+      const source = new VectorSource({
+        features: new GeoJSON().readFeatures(geojson, {
+          featureProjection: 'EPSG:3857', // TODO: revisit this
```

Comment:
This is something I'd like to hear thoughts on. I need to load the GeoJSON and provide a target projection, but I likely don't know the map projection at this this. So this is hardcoded to the default, which is not ideal.

Reply:
Resolved via https://github.com/openlayers/openlayers/pull/14927#discussion_r1282188137